pkg/cosign/tuf/client.go

```diff
@@ -105,6 +108,12 @@ type remoteCache struct {
 	Mirror string `json:"mirror"`
 }
 
+func resetForTests() {
+	singletonTUFMu.Lock()
```

Comment:
Since some of the tests are expecting a 'fresh' TUF setup, I added this to make sure we don't get any errors from modifying it without a lock in place.

Reply:
Tests within the tuf pkg will be sequential, and  we need the reset (but as you said, don't need the lock).

There is 1 test outside the TUF package that uses a custom TUF root. I'm concerned about that, that may be run in parallel. I will be refactoring that anyway to use a custom `CheckOpts` for the Rekor key rather than a custom TUF set up.

In the interim, I need to think how to lock that test.

Reply:
OK resolution: I added a TODO for that test that uses a custom root outside this package, expect a PR this afternoon addressing that.